src/Spice86.Core/Emulator/Memory/Memory.cs

```diff
@@ -23,15 +21,22 @@ public class Memory : Indexable.Indexable, IMemory {
     /// <summary>
     /// Instantiate a new memory bus.
     /// </summary>
+    /// <param name="memoryBreakpoints">The class that holds breakpoints based on memory access.</param>
     /// <param name="baseMemory">The memory device that should provide the default memory implementation</param>
-    /// <param name="is20ThAddressLineSilenced">whether the A20 gate is silenced.</param>
-    public Memory(IMemoryDevice baseMemory, bool is20ThAddressLineSilenced) {
+    /// <param name="a20gate">The class that implements A20 Gate on/off support.</param>
+    /// <param name="initializeResetVector">Whether to initialize the reset vector with a HLT instruction.</param>
+    public Memory(MemoryBreakpoints memoryBreakpoints, IMemoryDevice baseMemory, A20Gate a20gate, bool initializeResetVector = false) {
+        _memoryBreakpoints = memoryBreakpoints;
         uint memorySize = baseMemory.Size;
         _memoryDevices = new IMemoryDevice[memorySize];
-        Ram = new Ram(memorySize);
+        Ram = baseMemory;
         RegisterMapping(0, memorySize, Ram);
         (UInt8, UInt16, UInt16BigEndian, UInt32, Int8, Int16, Int32, SegmentedAddress) = InstantiateIndexersFromByteReaderWriter(this);
-        A20Gate = new(is20ThAddressLineSilenced);
+        A20Gate = a20gate;
+        if (initializeResetVector) {
```

Comment:
Shouldnt this be done by the bios?

Reply:
Yes, I don;t see this as a Memory responsibility.

Reply:
This is done bi the BIOS now.